pos_pricelist_show_discount/static/src/js/models.js

```diff
@@ -0,0 +1,91 @@
+odoo.define("pos_pricelist_show_discount.models", function (require) {
+    "use strict";
+    var models = require("point_of_sale.models");
+
+    var order_super = models.Order.prototype;
+    models.Order = models.Order.extend({
+        initialize: function () {
+            order_super.initialize.apply(this, arguments);
+            const self = this;
+            const config = self.pos.config;
+            if (
+                config.use_pricelist === true &&
+                config.display_discount_from_pricelist === true &&
+                config.discount_pricelist_id
+            ) {
+                self.pos.discount_pricelist = _.find(self.pos.pricelists, function (
+                    pricelist
+                ) {
+                    return pricelist.id === config.discount_pricelist_id[0];
+                });
+            } else {
+                self.pos.discount_pricelist = self.pos.default_pricelist;
+            }
+        },
+    });
+
+    var orderline_super = models.Orderline.prototype;
+    models.Orderline = models.Orderline.extend({
```

Comment:
i'd export:

```
 this.get_discount_pricelist_str();
 this.get_taxed_lst_unit_price();

``` 
for JSON and for printing as well, kind of:

```
  data.get_discount_pricelist_str = this.get_discount_pricelist_str();
  data.taxed_lst_unit_price = this.get_taxed_lst_unit_price();
```

so will be possible to print in receipt, if needed.

Reply:
afaik un need to set on `export_for_printing`, `export_for_JSON`, so that you i'll be able to have the value in the `env` for printing.

Reply:
Ah, I see now, thanks!